File: r/tests/test-yamlstar.R
# Copyright 2023-2026 Ingy dot Net
# This code is licensed under MIT license (See License for details)

# Test the yamlstar R binding.
# Run with: Rscript tests/test-yamlstar.R

library(yamlstar)

fails <- 0

check <- function(cond, label) {
  if (isTRUE(cond)) {
    cat("ok -", label, "\n")
  } else {
    cat("not ok -", label, "\n")
    fails <<- fails + 1
  }
}

# Load YAML mapping:
data <- yamlstar_load("test: 42")
check(data$test == 42, "load mapping")

# Load plain YAML:
data <- yamlstar_load("foo: bar")
check(data$foo == "bar", "load plain yaml")

# Load invalid input raises:
threw <- tryCatch(
  {
    yamlstar_load(":")
    FALSE
  },
  error = function(e) TRUE
)
check(threw, "load error raises")

# Load multiple times:
data <- yamlstar_load("test: 42")
check(data$test == 42, "load multiple times")

if (fails > 0) {
  cat(fails, "test(s) failed\n")
  quit(status = 1)
}

File: nim/tests/test_yamlstar.nim
# Copyright 2023-2026 Ingy dot Net
# This code is licensed under MIT license (See License for details)

# Test the yamlstar Nim binding.

import std/json

import yamlstar

var fails = 0

proc check(cond: bool, label: string) =
  if cond:
    echo "ok - ", label
  else:
    echo "not ok - ", label
    inc fails

let yaml = newYAMLStar()

# Load YAML mapping:
var data = yaml.load("test: 42")
check(data["test"].getInt == 42, "load mapping")

# Load plain YAML:
data = yaml.load("foo: bar")
check(data["foo"].getStr == "bar", "load plain yaml")

# Load invalid input raises and sets error:
var threw = false
try:
  discard yaml.load("key: \"unclosed")
except YAMLStarError:
  threw = true
check(threw, "load error raises")
check(yaml.error != nil, "error object is set")

# Load multiple times on one instance:
data = yaml.load("test: 42")
check(data["test"].getInt == 42, "load multiple times")

yaml.close()

if fails > 0:
  echo fails, " test(s) failed"
  quit(1)
